x-pack/plugins/apm/server/routes/storage_explorer/get_storage_details_per_service.ts

```diff
@@ -153,3 +155,122 @@ export async function getStorageDetailsPerProcessorEvent({
     };
   });
 }
+
+export async function getStorageDetailsPerIndex({
+  setup,
+  context,
+  indexLifecyclePhase,
+  randomSampler,
+  start,
+  end,
+  environment,
+  kuery,
+  serviceName,
+}: {
+  setup: Setup;
+  context: ApmPluginRequestHandlerContext;
+  indexLifecyclePhase: IndexLifecyclePhaseSelectOption;
+  randomSampler: RandomSampler;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+  serviceName: string;
+}) {
+  const { apmEventClient } = setup;
+
+  const [
+    { indices: allIndicesStats },
+    indicesLifecycleStatus,
+    indicesInfo,
+    response,
+  ] = await Promise.all([
+    getTotalIndicesStats({ setup, context }),
+    getIndicesLifecycleStatus({ setup, context }),
+    getIndicesInfo({ setup, context }),
+    apmEventClient.search('get_storage_details_per_index', {
```

Comment:
It might reduce the size of the function but can get more confusing as we already have 3 indices-related functions here `getTotalIndicesStats`, `getIndicesLifecycleStatus`, `getIndicesInfo`.